### PR TITLE
Loginlogout

### PR DESCRIPTION
**Pull request by jvssicawu**

### CODE REVIEW

File: imports/redux/reducers/index.jsx [hunk 1]
```diff
@@ -95,6 +75,33 @@ function userReducer(state = initialUserState, action) {
       break; //TODO: remove after reducer refactor
     case LOGOUT_USER:
       return { ...state, isSignedIn: false, fullName: "", email: "", blacklist: [], favourites: [] };
+    case SIGNUP_USER:
+      let query = UserInfo.find({email: action.email}).fetch();
+      let userExists = query[0];
+      if (userExists) {
+        alert("An account with this email already exists. Proceed to login to continue.");
```
Comment (hailin-zhang) [inline imports/redux/reducers/index.jsx]: As a side note, doing this is generally a security flaw. We don't want users to know what's in our database!

File: imports/redux/reducers/index.jsx [hunk 2]
```diff
@@ -95,6 +75,33 @@ function userReducer(state = initialUserState, action) {
...
+      else {
+        UserInfo.insert({
+          email: action.email,
+          firstName: action.firstName,
+          lastName: action.lastName,
+          password: action.password,
+          preferences: {blacklist: [], favourites: []}
+        }, function(err) {
+          if (err) {
+            console.log(err);
```
Comment (hailin-zhang) [inline imports/redux/reducers/index.jsx]: do we need this console log?

File: imports/redux/reducers/index.jsx [hunk 3]
```diff
@@ -95,6 +75,33 @@ function userReducer(state = initialUserState, action) {
...
+          }
+        });
+        return { ...state,
+          email: action.email,
+          isSignedIn: true,
+          fullName: action.firstName + " " + action.lastName,
```
Comment (hailin-zhang) [inline imports/redux/reducers/index.jsx]: don't use string concatenation. use interpolation instead

File: imports/redux/reducers/index.jsx [hunk 4]
```diff
@@ -95,6 +75,33 @@ function userReducer(state = initialUserState, action) {
...
+          firstName: action.firstName,
+          lastName: action.lastName,
+          password: action.password,
+          preferences: {blacklist: [], favourites: []}
+        }, function(err) {
+          if (err) {
+            console.log(err);
+          }
+        });
+        return { ...state,
```
Comment (hailin-zhang) [inline imports/redux/reducers/index.jsx]: move this to next line

File: imports/ui/shared_components/signup/signup.jsx [hunk 5]
```diff
@@ -46,73 +48,101 @@ const DialogTitle = withStyles(styles)((props) => {
...
 }))(MuiDialogActions);
 
 
 class Signup extends React.Component {
+  constructor(props){
+    super(props);
+    this.firstName = React.createRef();
+    this.lastName = React.createRef();
+    this.email = React.createRef();
+    this.password = React.createRef();
```
Comment (hailin-zhang) [inline imports/ui/shared_components/signup/signup.jsx]: side note: consider hashing this

Comment (js1998) [reply imports/ui/shared_components/signup/signup.jsx]: I'll make an issue

**Comment by js1998** [comment]
This PR has been added as part of PR #46

**Pull request merged**